### PR TITLE
remove backfill from eip 2935 and serve hashes from state when eip 7709 enabled

### PR DESCRIPTION
**Pull request by yerke26**
Eip2935 is going to collect block hashes in the state, when Eip7709 gets enabled it will start serving collected block hashes from the state. 

Before: if eip2935IsEnabled we stored the last 8192 hashes in the state and served last 256 block hashes from state on request
In this PR: if eip2935IsEnabled we store parent block hash in the state and if eip7709IsEnabled serve last 256 block hashes from state

#### What types of changes does your code introduce?

- [ ] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [x] Optimization
- [ ] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [x] Yes
- [ ] No

#### If yes, did you write tests?

- [ ] Yes
- [ ] No

#### Notes on testing
There is no tests for Eip7709, tests that confirm blockhashes being served from the state
Eip2935 tests cannot test that blockhashes stored in the state

_Optional. Remove if not applicable._

#### Requires documentation update

- [ ] Yes
- [ ] No

_If yes, link the PR to the docs update or the issue with the details labeled `docs`. Remove if not applicable._

#### Requires explanation in Release Notes

- [ ] Yes
- [ ] No

_If yes, fill in the details here. Remove if not applicable._

## Remarks

_Optional. Remove if not applicable._

### CODE REVIEW

File: src/Nethermind/Nethermind.Blockchain/Blocks/BlockhashStore.cs [hunk 1]
```diff
@@ -47,28 +39,12 @@ public void ApplyHistoryBlockHashes(BlockHeader blockHeader)
             return null;
         }
         var blockIndex = new UInt256((ulong)(requiredBlockNumber % Eip2935Constants.RingBufferSize));
-        Address? eip2935Account = spec.Eip2935ContractAddress ?? Eip2935Constants.BlockHashHistoryAddress;
+        Address eip2935Account = spec.Eip2935ContractAddress;
```
Comment (tanishqjasoria) [inline src/Nethermind/Nethermind.Blockchain/Blocks/BlockhashStore.cs:42]: this should not be removed

**Comment by tanishqjasoria** [comment]
NOTE: though 7709 is included here but the implementation is not complete yet -  we dont charge the appropiate gas as its dependent on verkle. 
Should we keep it or just remove it entirely?
cc: @MarekM25 @LukaszRozmej

File: src/Nethermind/Nethermind.Evm.Test/Eip2935Tests.cs [hunk 2]
```diff
@@ -28,7 +28,7 @@ public override void Setup()
 
 
     [TestCase(MainnetSpecProvider.CancunBlockTimestamp, false)]
-    [TestCase(MainnetSpecProvider.PragueBlockTimestamp, true)]
+    [TestCase(MainnetSpecProvider.PragueBlockTimestamp, false)]
     public void CorrectBlockhashBeingUsed(ulong timestamp, bool eipEnabled)
```
Comment (tanishqjasoria) [inline src/Nethermind/Nethermind.Evm.Test/Eip2935Tests.cs:32]: here eipEnabled=false does not make sense
eip2935 is enabled - just the expectation changed
i think this test can be removed or modify it for using 7709

File: src/Nethermind/Nethermind.Blockchain.Test/BlockhashProviderTests.cs [hunk 3]
```diff
@@ -277,14 +277,14 @@ public void Eip2935_init_block_history_and_then_get_hash(int chainLength)
...
         }
 
-        private static void AssertGenesisHash(BlockhashProvider provider, BlockHeader currentHeader,
-            Hash256 genesisHash)
+        private static void AssertGenesisHash(IReleaseSpec spec, BlockhashProvider provider, BlockHeader currentHeader,
+            Hash256? genesisHash)
         {
             Hash256? result = provider.GetBlockhash(currentHeader, 0);
-            if (currentHeader.Number > Eip2935Constants.RingBufferSize)
+            if (currentHeader.Number > Eip2935Constants.RingBufferSize || spec.IsEip7709Enabled || currentHeader.Number > 256)
```
Comment (tanishqjasoria) [inline src/Nethermind/Nethermind.Blockchain.Test/BlockhashProviderTests.cs:287]: this condition does not make sense - i think this should be
`(spec.IsEip7709Enabled && currentHeader.Number > Eip2935Constants.RingBufferSize) || currentHeader.Number > 256`

**Review by tanishqjasoria — CHANGES_REQUESTED** [review]

File: src/Nethermind/Nethermind.Blockchain/Blocks/BlockhashStore.cs (hunk 1 above)
Comment (tanishqjasoria) [reply src/Nethermind/Nethermind.Blockchain/Blocks/BlockhashStore.cs:42]: @yerke26

**Review by tanishqjasoria — APPROVED** [review]

**Review by rjnrohit — APPROVED** [review]

**Review by LukaszRozmej — APPROVED** [review]

**Pull request merged**